node/runtime/src/lib.rs

```diff
@@ -871,6 +908,15 @@ impl_runtime_apis! {
 			NativeTokenManagement::initialized()
 		}
 	}
+
+	impl sp_block_production_log::BlockProductionLogApi<Block, CommitteeMember<CrossChainPublic, SessionKeys>>  for Runtime {
+		fn get_current_author() -> CommitteeMember<CrossChainPublic, SessionKeys> {
+			let mut committee = SessionCommitteeManagement::get_current_committee().1;
+			let slot = pallet_aura::CurrentSlot::<Runtime>::get();
+			// this implementation is compatible with round-robin employed by Aura
+			committee.remove(*slot as usize % committee.len())
+		}
+	}
```

Comment:
I am not sure regarding the business logic, but it looks like we are fetching committee which is a vector which involves the copy of the underlying structure, but later we only take one element of if which we return. 
Maybe it will be good idea to implement some function for `SessionCommitteeManagement` that will make this single lookup for us

Reply:
or we can use `SessionCommitteeManagement::get_current_authority(slot)`, please check if it is doing the same

Reply:
also once we will be there, we can adjust the name of the function, maybe `current_authority` will be more idiomatic
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Reply:
Yeah, I don't want to change that. I don't see a way to get the committee size and get nth member modulo that size with one storage read, unless I move round-robin logic into the pallet and I don't think it belongs there.

Reply:
Yep, but we do not need to change the logic too much. We can use something like this with maybe some comment that we make a boundary check with modulo operation:
```suggestion
		fn get_current_author() -> CommitteeMember<CrossChainPublic, SessionKeys> {
			let committee = SessionCommitteeManagement::get_current_committee().1;
			let slot = pallet_aura::CurrentSlot::<Runtime>::get();
			// this implementation is compatible with round-robin employed by Aura
			committee.get_unchecked(*slot as usize % committee.len())
		}
	}
```
or for example use [swap_remove](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap_remove), which is O(1). It will reorder elements, but we drop this vec at the end of the function, so there will be no impact, but it may be less readable

Reply:
I'll remove the `remove` tho